assets/js/components/App.tsx

```diff
@@ -30,6 +31,19 @@ class AppRoutes extends React.Component {
 }
 
 const App = (): ReactElement<HTMLDivElement> => {
+  const sentryDsn = document.getElementById("app")?.dataset.sentry;
+  const username = document.getElementById("app")?.dataset.username;
+
+  if (sentryDsn) {
+    Sentry.init({
+      dsn: sentryDsn,
+    });
+
+    if (username) {
+      Sentry.setUser({ username: username });
+    }
+  }
```

Comment:
Slightly worried about putting this right in the render logic of a React component, since if it ever re-renders, these lines will run again. I don't think the top-level component will ever re-render, but still...

Unless this is exactly what the Sentry docs recommend as a best practice, would you mind doing one of the following?
1. Move this into a `useEffect` with `[]` as its dependency list, or
2. Call the Sentry initialization logic at the module level, in this module, right after the import statements.